src/messages/variant.rs

```diff
@@ -49,6 +49,11 @@ pub(crate) enum Variant {
         shared_state: SharedState,
         parsec_version: u64,
     },
+    /// Message sent to a lagging peer.
+    NotifyLagging {
+        shared_state: SharedState,
+        parsec_version: u64,
```

Comment:
Why parsec_version if that's what we are removing?  I think this field can be removed completely. All we need send the lagging peer is the `ProofChain` he needs plus current elders AFIAK.

Reply:
> Why parsec_version if that's what we are removing?

This just to be consistent with other variant (and the handling logic as well). I think it will be easy for the `parsec removing` PR to remove all this stuff at once.

> All we need send the lagging peer is the ProofChain he needs plus current elders

SharedState is almost about this, especially there is further plan to remove its `relocate_queue` further.

Reply:
I feel we should make the consistency removing this from the other variant. It would make more sense I feel.

Reply:
Well, this is bit like a chicken-egg issue: this PR is supposed to be a pre-condition of removal parsec, and consistency of removing parsec_version will mean remove of parsec.

I do feel it will be better have this PR contain this, and remove all in a later on PR.

Reply:
Ok then I suppose we can get the remove parsec done very soon and this can stop hurting our eyes :-)